specification/agricultureplatform/AgriculturePlatform.DataPlane/routes.tsp

```diff
@@ -0,0 +1,512 @@
+
+      /**
+       * Filters on key-value pairs within the Properties object.
+       * eg. "{testKey} eq {testValue}".
+       */
+      #suppress "@azure-tools/typespec-azure-core/no-query-explode" ""
+      @query(#{ name: "propertyFilters", explode: true })
+      propertyFilters?: string[];
+
+      /**
+       * Statuses of the resource.
+       */
+      #suppress "@azure-tools/typespec-azure-core/no-query-explode" ""
+      @query(#{ name: "statuses", explode: true })
+      statuses?: string[];
+
+      /**
+       * Ids of the resource.
+       */
+      #suppress "@azure-tools/typespec-azure-core/no-query-explode" ""
+      @query(#{ name: "ids", explode: true })
+      ids?: string[];
+
+      /**
+       * Minimum creation date of resource (inclusive).
+       */
+      @query("minCreatedDateTime")
+      minCreatedDateTime?: utcDateTime;
+
+      /**
+       * Maximum creation date of resource (inclusive).
+       */
+      @query("maxCreatedDateTime")
+      maxCreatedDateTime?: utcDateTime;
+
+      /**
+       * Minimum last modified date of resource (inclusive).
+       */
+      @query("minLastModifiedDateTime")
+      minLastModifiedDateTime?: utcDateTime;
+
+      /**
+       * Maximum last modified date of resource (inclusive).
+       */
+      @query("maxLastModifiedDateTime")
+      maxLastModifiedDateTime?: utcDateTime;
+
+      /**
+       * Maximum number of items needed (inclusive).
+       * Minimum = 10, Maximum = 1000, Default value = 50.
+       */
+      @maxValue(1000)
+      @minValue(10)
+      @query("maxPageSize")
+      maxPageSize?: int32 = 50;
+
+      /**
+       * Skip token for getting next set of results.
+       */
+      @query("skipToken")
+      skipToken?: string;
+    }>
+  >;
+
+  /**
+   * Gets a specified Collection resource.
+   */
+  get is Operations.ResourceRead<Collection>;
+
+  /**
+   * Creates or updates a Collection resource.
+   */
+  createOrUpdate is Operations.ResourceCreateOrUpdate<Collection>;
+
+  /**
+   * Deletes Collection for given Collection id.
+   */
+  delete is Operations.ResourceDelete<Collection>;
+}
+
+interface FileOperations {
+  /**
+   * Deletes file.
+   */
+  @action("delete")
+  delete is Operations.ResourceCollectionAction<
+    FileResource,
+    File,
+    NoContentResponse
+  >;
+
+  /**
+   * Downloads and returns file Stream as response for the given input filePath.
+   */
+  @action("download")
+  download is Operations.ResourceCollectionAction<
+    FileResource,
+    File,
+    NoContentResponse,
+    RequestHeadersTrait<{
+      /**
+       * Accept header
+       */
+      @header
+      accept: "application/json, application/octet-stream";
+    }>
+  >;
+
+  /**
+   * Uploads a file.
+   */
+  @action("upload")
+  upload is Operations.ResourceCollectionAction<FileResource, {}, File>;
+}
+
+interface ItemsOperations {
+  /**
+   * Creates or updates a Item resource.
+   */
+  @action("create")
+  create is Operations.ResourceCollectionAction<
```

Comment:
Can you try using https://azure.github.io/typespec-azure/docs/libraries/azure-core/reference/interfaces/#Azure.Core.ResourceCreateWithServiceProvidedName? If so, then you can probably also delete the `ItemResponse` model